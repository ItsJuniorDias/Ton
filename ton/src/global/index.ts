export * from './shadowStyles';
export * from './theme';
