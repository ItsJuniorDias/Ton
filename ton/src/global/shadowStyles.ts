export const shadowStyles = {
  shadowColor: '#000',
  shadowOffset: {
    width: 0,
    height: 5,
  },
  shadowOpacity: 0.1,
  shadowRadius: 2.62,
  elevation: 2,
};
