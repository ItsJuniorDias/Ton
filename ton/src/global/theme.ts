export const theme = {
  colors: {
    primary: '#01ad0b',
    secondary: '#01ea34',
    shape: '#ffffff',
    background: '#eff4f8',
    title: '#000000',
    text: '#20252b',
    notification: '#ff7b00',
    alert: '#ed5e69',
  },
  fonts: {
    regular: 'Roboto-Regular',
    medium: 'Roboto-Medium',
    bold: 'Roboto-Bold',
  },
};
