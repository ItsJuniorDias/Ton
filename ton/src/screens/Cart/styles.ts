import styled from 'styled-components/native';
import { scale } from '../../utils';

export const Content = styled.View`
  flex: 1;
  padding: 0px 24px;
  padding-bottom: 32px;
  background-color: ${({ theme }) => theme.colors.background};
`;
