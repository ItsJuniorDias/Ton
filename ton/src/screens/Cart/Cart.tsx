import React from 'react';
import { FlatList } from 'react-native';
import { Header, ItemCart } from '../../components';
import { useAppSelector } from '../../hooks';

import { Content, TitleHeaderCart } from './styles';

export const Cart = () => {
  const cart = useAppSelector((state) => state.cart);

  return (
    <>
      <Header title="Cart" amountCart={cart.data.length} isGoBack />

      <Content>
        <FlatList
          data={cart.data}
          renderItem={({ item }) => <ItemCart {...item} />}
          ListHeaderComponent={() => (
            <TitleHeaderCart>
              {cart.data.length} produtos adicionados
            </TitleHeaderCart>
          )}
          keyExtractor={(item) => item.id}
        />
      </Content>
    </>
  );
};
