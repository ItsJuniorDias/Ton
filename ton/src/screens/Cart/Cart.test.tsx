import 'jest-styled-components';
import React from 'react';
import { render } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';
import { NavigationContainer } from '@react-navigation/native';
import { Cart } from './Cart';

import { theme } from '../../global';
import { Provider } from 'react-redux';
import store from '../../store';

jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon');
jest.mock('react-native-iphone-x-helper', () => ({
  getStatusBarHeight: jest.fn(),
  isIphoneX: jest.fn(),
}));

describe('Behavior Cart', () => {
  const screenRender = () => (
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <NavigationContainer>
          <Cart />
        </NavigationContainer>
      </ThemeProvider>
    </Provider>
  );

  it('render snapshot', () => {
    const result = render(screenRender()).toJSON();

    expect(result).toMatchSnapshot();
  });
});
