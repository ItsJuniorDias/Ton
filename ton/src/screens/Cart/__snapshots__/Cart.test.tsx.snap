// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Behavior Cart render snapshot 1`] = `
[
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "flexDirection": "row",
          "height": 188,
          "justifyContent": "space-between",
          "paddingBottom": 0,
          "paddingLeft": 47,
          "paddingRight": 47,
          "paddingTop": "px",
          "width": "100%",
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "opacity": 1,
          },
        ]
      }
      testID="pressableHeader_testID"
    >
      <Icon
        color="#20252b"
        name="arrow-back-ios"
        size={22}
      />
    </View>
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "Roboto-Medium",
            "fontSize": 35,
          },
        ]
      }
    >
      Cart
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "opacity": 1,
          },
        ]
      }
      testID="pressableCart_testID"
    >
      <Icon
        color="#01ad0b"
        name="shopping-cart"
        size={28}
      />
      <View
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#ff7b00",
              "borderRadius": 20,
              "height": 35,
              "justifyContent": "center",
              "marginLeft": 31,
              "marginTop": -16,
              "position": "absolute",
              "width": 35,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "Roboto-Regular",
                "fontSize": 27,
              },
            ]
          }
          testID="amountCart_testID"
        >
          2
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      [
        {
          "backgroundColor": "#eff4f8",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingBottom": 32,
          "paddingLeft": 24,
          "paddingRight": 24,
          "paddingTop": 0,
        },
      ]
    }
  >
    <RCTScrollView
      data={
        [
          {
            "id": "T1",
            "price": 118.8,
            "thumbnail": "https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_256,q_auto/site-ton/t1-1",
            "title": "T1 Brother",
          },
          {
            "id": "T1_CHIP",
            "price": 190.8,
            "thumbnail": "https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_256,q_auto/site-ton/t1-chip-1",
            "title": "T1 Chip Brother",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "borderRadius": 31,
                  "flexDirection": "row",
                  "height": 188,
                  "justifyContent": "space-between",
                  "marginTop": 63,
                  "paddingBottom": 31,
                  "paddingLeft": 31,
                  "paddingRight": 31,
                  "paddingTop": 31,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  },
                ]
              }
            >
              <Image
                resizeMode="contain"
                source={
                  {
                    "uri": "https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_256,q_auto/site-ton/t1-1",
                  }
                }
                style={
                  [
                    {
                      "height": 126,
                      "width": 126,
                    },
                  ]
                }
              />
              <Text
                style={
                  [
                    {
                      "color": "#20252b",
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 31,
                    },
                  ]
                }
              >
                T1 Brother
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "opacity": 1,
                  },
                ]
              }
              testID="pressableItemCart_testID"
            >
              <Icon
                color="#ed5e69"
                name="highlight-remove"
                size={28}
              />
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "borderRadius": 31,
                  "flexDirection": "row",
                  "height": 188,
                  "justifyContent": "space-between",
                  "marginTop": 63,
                  "paddingBottom": 31,
                  "paddingLeft": 31,
                  "paddingRight": 31,
                  "paddingTop": 31,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  },
                ]
              }
            >
              <Image
                resizeMode="contain"
                source={
                  {
                    "uri": "https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_256,q_auto/site-ton/t1-chip-1",
                  }
                }
                style={
                  [
                    {
                      "height": 126,
                      "width": 126,
                    },
                  ]
                }
              />
              <Text
                style={
                  [
                    {
                      "color": "#20252b",
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 31,
                    },
                  ]
                }
              >
                T1 Chip Brother
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "opacity": 1,
                  },
                ]
              }
              testID="pressableItemCart_testID"
            >
              <Icon
                color="#ed5e69"
                name="highlight-remove"
                size={28}
              />
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>,
]
`;
